ext/exif/exif.c

```diff
@@ -215,6 +215,26 @@ zend_module_entry exif_module_entry = {
 ZEND_GET_MODULE(exif)
 #endif
 
+/* php_stream_read() may return early without reading all data, depending on the chunk size
+ * and whether it's a URL stream or not. This helper keeps reading until the requested amount
+ * is read or until there is no more data available to read. */
+/* FIXME: perhaps this should be moved to some more general place? */
+static ssize_t read_from_stream_looped(php_stream *stream, char *buf, size_t count)
```

Comment:
I would maybe rename it to make clear that this can work only for stream files so in case someone decides extend exif, it is clear that this is not gonna work correctly for some other stream types.